operator/v1/types_console.go

```diff
@@ -143,8 +143,127 @@ type Capability struct {
+	// When "Dark" is provided, the corresponding logo for the dark theme will be used in the console UI.
+	// When "Light" is provided, the corresponding logo for the light theme will be used in the console UI.
+	// +kubebuilder:validation:Enum="Dark";"Light"
+	// +required
+	Type ThemeType `json:"type"`
+
+	// refKey is the key pointing to a specific key/value inside the ConfigMap.
+	// This field is an enum with valid values "MastheadDark", "MastheadLight", "FaviconDark", and "FaviconLight".
+	// When "MastheadDark" is provided in the ConfigMap, the referenced logo for the dark theme will be used in the masthead of the console UI.
+	// When "MastheadLight" is provided in the ConfigMap, the referenced logo for the light theme will be used in the masthead of the console UI.
+	// When "FaviconDark" is provided in the ConfigMap, the referenced logo for the dark theme will be used in the favicon of the console UI.
+	// When "FaviconLight" is provided in the ConfigMap, the referenced logo for the light theme will be used in the favicon of the console UI.
+	// +kubebuilder:validation:Enum="MastheadDark";"MastheadLight";"FaviconDark";"FaviconLight"
+	// +required
+	RefKey RefKeyType `json:"refKey"`
+}
+
+// CustomLogo defines a configuration based on theme types for the console UI logo.
+type CustomLogo struct {
+	// type specifies the type of the logo for the console UI. It determines whether the logo is for the masthead or favicon.
+	// This field is an enum with valid values "Masthead" and "Favicon".
+	// Valid values:
+	// - "Masthead": When this value is provided, the logo will be used in the masthead or about modal of the console UI.
+	// - "Favicon": When this value is provided, the logo will be used as the favicon of the console UI.
+	// +kubebuilder:validation:Enum="Masthead";"Favicon"
+	// +required
+	Type LogoType `json:"type"`
+
+	// themes specifies the themes for the console UI logo.
+	// This field is a list of themes, each with a Type and a RefName.
+	// Each theme determines whether the logo is for the dark or light theme of the console UI.
+	// The valid values for the Type field are "Dark" and "Light".
+	// The RefName field should reference the name of the ConfigMap that contains the logo files.
+	// Each item in the list must have a unique combination of Type and RefName.
+	// +kubebuilder:validation:Enum="Dark";"Light"
+	// +kubebuilder:validation:MinItems=1
+	// +kubebuilder:validation:MaxItems=2
+	// +listType=map
+	// +listMapKey=type
+	// +required
+	Themes []Theme `json:"themes"`
+
+	// refName is the reference to a specific ConfigMap name in the openshift-config namespace containing custom logo files.
+	// The ConfigMap must be created with the appropriate keys and file extensions to ensure the console serves the file with the correct MIME type.
+	// The required keys are "MastheadDark", "MastheadLight", "FaviconDark", and "FaviconLight".
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=253
```

Comment:
Because this is supposed to be a ConfigMap name reference, are there any other validations that should be put in place? Maybe use the same name validation that Kubernetes itself uses for validating configmap names?